src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -2051,5 +1995,55 @@ string[] linkingTargetFields
                 RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig);
             }
         }
+
+        /// <summary>
+        /// Validates that the runtime base-route is well-formatted and does not contain any reserved characeters and
+        /// can only be configured when authentication provider is Static Web Apps.
+        /// </summary>
+        /// <param name="runtimeBaseRoute">Value of runtime base-route as specified in config.</param>
+        /// <param name="authenticationProvider">The authentication provider configured.</param>
+        /// <param name="isExceptionExpected">Whether an exception is expected as a result of test run.</param>
+        /// <param name="expectedExceptionMessage">The expected exception message.</param>
+        [DataTestMethod]
+        [DataRow("/base-route", "StaticWebApps", false, DisplayName = "Runtime base-route correctly configured for Static Web Apps.")]
+        [DataRow("/base-route", "AppService", true, "Runtime base-route is not used for non-Static Web Apps authentication providers.",
+            DisplayName = "Runtime base-route specified for non-Static Web Apps authentication provider - AppService.")]
+        [DataRow("/base+?route", "StaticWebApps", true, $"Runtime base-route {RuntimeConfigValidator.URI_COMPONENT_WITH_RESERVED_CHARS_ERR_MSG}",
+            DisplayName = "Runtime base-route specified for Static Web Apps authentication provider containing reserved characters +?")]
+        [DataRow("/base%&#route", "StaticWebApps", true, $"Runtime base-route {RuntimeConfigValidator.URI_COMPONENT_WITH_RESERVED_CHARS_ERR_MSG}",
+            DisplayName = "Runtime base-route specified for Static Web Apps authentication provider containing reserved characters %&#")]
+        [DataRow("base-route", "StaticWebApps", true, $"Runtime base-route should start with a '/'.",
+            DisplayName = "Runtime base-route specified for Static Web Apps authentication provider not starting with '/'")]
+        public void ValidateRuntimeBaseRouteSettings(
```

Comment:
May be worth having two additional tests to show that BaseRoute being null is handled properly :
1. base-route value is `null` and provider is non-EasyAuth ~ "otherProvider"
2. base-route value is `null` and provider is StaticWebApps

Reply:
There is no difference in between the two cases. Essentially because we cannot differentiate between the scenarios when base-route is configured as null vs when base-route is not specified at all. So, when the auth provider is not Static Web Apps, and base-route is defined as null, we don't throw an exception.

Reply:
Only reason i bring this up is to add coverage for the case that Ani is fixing in #1580. 

Your point is valid that validation only occurs if a value is actually provided. Wouldn't a null case here validate that statement?

Reply:
But when we see that base-route is null in runtimeconfigvalidator, we can't tell if it is null because of it being absent in the config or being explicitly specified as null. We can do something like this - we can mandate that if the base-route is specified, it has to be a non-empty string.

Reply:
> "we can't tell if it is null because of it being absent in the config or being explicitly specified as null" 

That part is fine. 

Without a null test case, do we have coverage that explicitly validates that "no unexpected errors or validations occur if `base-path` is null?"

> we can mandate that if the base-route is specified, it has to be a non-empty string.

Probably would want to go further than just empty string and disallow a whitespace only base-route. You may want to update `if (!string.IsNullOrEmpty(runtimeBaseRoute))` to `if (!string.IsNullOrWhitespace(runtimeBaseRoute))` in runtimeconfigvalidator and add a test to confirm behavior.

Reply:
Well actually, we should continue with the current scenario. We should be allowed to specify null/whitespace base-route. Even though right now only SWA will use this base-route feature with value "/data-api" but we shouldn't put any restriction. We would just not append base-route in the uri if it is null/whitespace.

Reply:
> Well actually, we should continue with the current scenario. We should be allowed to specify null/whitespace base-route.

I agree that we should not append the base route in the url if base route is null. 

However, if the base route is whitespace, that implies that the end user intentionally provided a value for that property. If we allow that we should add a comment for 'why?' and also add public documentation that whitespace only will be ignored, even if it doesn't cause a validation failure.
- uri spec allows whitespace in url but the whitespace values will be escaped by browsers/and uri helper functions. Though a whitespace only base url is technically not forbidden by spec.. my sentiment earlier for why i initially suggested it shouldn't be allowed is that it doesn't make sense to me as a valid use case. If we want to disallow it in the future, we can revisit in a future pr.

Reply:
Let me correct myself. Base-route should not be appended if it is null. And if the user intentionally provides it as  string containing whitespaces (i.e. `/      `, we should throw an exception as this makes the url invalid.